lib/src/widgets/adw/new/header_button.dart

```diff
@@ -1,9 +1,10 @@
 import 'package:flutter/material.dart';
-import 'package:libadwaita/src/utils/colors.dart';
+import 'package:libadwaita/libadwaita.dart';
 
-enum ButtonStatus { normal, hover, tapDown }
-
-class AdwHeaderButton extends StatefulWidget {
+/// Button that is usually placed inside an `AdwHeaderBar` widget.
+///
+/// Via the `icon` parameter, a widget can be placed inside the button.
+class AdwHeaderButton extends StatelessWidget {
   /// The icon of the button. Default size would be `17`.
   final Widget icon;
```

Comment:
I would rename `icon` to `child` since it can be any widget, although I know it was named `icon` before so it's not your fault!

Reply:
I'd like to rename the `icon` parameter to `child`, makes sense.

I didn't want to introduce breaking changes in this PR, but I can create a future PR to resolve this.

Reply:
That sounds great.